Tests/BodegaTests/SQLiteStorageEngineTests.swift

```diff
@@ -195,6 +195,25 @@ final class SQLiteStorageEngineTests: XCTestCase {
         XCTAssertEqual(updatedKeyCount, 0)
     }
 
+    func testVacuumAfterRemoveAllData() async throws {
```

Comment:
This test is correct, but would you mind making a few changes to match the other tests?

1. I would prefer to use the `storage` property defined at the top of the file, rather than creating `otherStorage` below. 
2. I would also prefer to use the `writeItemsToDatabase` function rather than creating a for loop in line to write items to the database. You're also free to update the `writeItemsToDatabase` method to take in an SQLiteStorageEngine if that would help simplify the code.
3. I didn't test this and it's more of a nit but I think you can skip the explicit `: Int` type annotations to match the style elsewhere.

Reply:
The main problem with this test, as I wrote it, compared to the other tests, is that it requires the location of the sqlite3 file. However, since this location is private in `SQLiteStorageEngine`, I cannot access it in the test. Therefore, I have to create a new storage and store the location. Another possible solution, which may be better, is to store the temporary path of `storage` in the `SQLiteStorageEngineTests` class. Will change that.

Reply:
I've updated the test, and I think it looks much better now.